### PR TITLE
Search result and header nav front-end

### PR DESCRIPTION
**Pull request by delimac**
- Made sections stick on top when scrolling
- Navbar links conditionals when user is logged-in or in a different page

https://user-images.githubusercontent.com/37357478/119376571-afa50d00-bcee-11eb-9843-1cdc50b38fb4.mp4

### CODE REVIEW

**Review by saikatb221062 — APPROVED** [review]

File: app/views/shared/_nav-links.html.erb [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+      <%= link_to 'CONVENIENCE', root_path, class: 'text-dark pb-2' %>
+    <% end %>
+  </div>
+
+  <div>
+    <% if user_signed_in? %>
+      <%= link_to "LOG OUT", destroy_user_session_path, method: :delete, class: 'btn btn-dark rounded-pill ml-3 px-2 px-md-4' %>
+    <% else %>
+      <%= link_to 'LOG IN', new_user_session_path, class: 'btn btn-outline-dark mr-2 mr-md-3 rounded-pill px-2 px-md-4' %>
+      <%= link_to 'SIGN UP', new_user_registration_path, class: 'btn btn-dark rounded-pill px-2 px-md-4' %>
```
Comment (saikatb221062) [inline app/views/shared/_nav-links.html.erb:16]: Hi Chris - request a small addition after this line 
       <div class="d-none" id="current-user-email">
          <%= current_user.email %>
       </div>
I am keeping this as a no-display element, as I need it in my JS, to redirect to devise sign-in, if the user has clicked "Add to Cart" but has not signed in. My JS script uses a id selector to examine current-user-email and redirects page to devise, if it is null (nil).

File: app/views/shared/_nav-links.html.erb [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+    <% end %>
```
Comment (saikatb221062) [inline app/views/shared/_nav-links.html.erb:17]: \<div class="d-none" id="current-user-email">
          <%= current_user.email %>
        \</div>
Request Chris to add this after line 17, I am inserting user email as a no-display field, I need this element for my JS which redirects to devise sign-in, if the user has clicked "Add to Cart" but is not signed in .... thanks !

**Review by saikatb221062 — CHANGES_REQUESTED** [review]

File: app/views/shared/_nav-links.html.erb (hunk 2 above)
Comment (delimac) [reply app/views/shared/_nav-links.html.erb:17]: Hey @saikatb221062, Ive added the requested code and pushed to the branch.

**Comment by miny0ungchang** [comment]
@delimac looks good, please merge

**Pull request merged**